pkg/apiserver/handlers.go

```diff
@@ -340,22 +333,20 @@ func (r *APIRequestInfoResolver) GetAPIRequestInfo(req *http.Request) (APIReques
 
 	// URL forms: /namespaces/{namespace}/{kind}/*, where parts are adjusted to be relative to kind
 	if currentParts[0] == "namespaces" {
-		if len(currentParts) < 3 {
-			requestInfo.Resource = "namespaces"
-			if len(currentParts) > 1 {
-				requestInfo.Namespace = currentParts[1]
-			}
-		} else {
-			requestInfo.Resource = currentParts[2]
+		if len(currentParts) > 1 {
 			requestInfo.Namespace = currentParts[1]
-			currentParts = currentParts[2:]
+
+			// if there is another step after the namespace name and it is not a known namespace subresource
+			// move currentParts to include it as a resource in its own right
+			if len(currentParts) > 2 {
```

Comment:
Probably `/namespaces/foo` should also return the resources included in that namespace.  Or we could explicitly handle `/namespaces/foo/` with the extra slash on the end.

The advantage of the current path setup is that "DELETE /namespaces/foo" results in "/namespaces/foo/*" no longer existing.